src/index.ts

```diff
@@ -357,6 +385,7 @@ const main = async () => {
     {
       dqlStatement: z.string(),
     },
+    { readOnlyHint: true },
```

Comment:
I would also consider this idempotent. If a DQL is (syntactically) valid, it will also be equally valid in a month; likewise if it's invalid.

Reply:
agree